components/tabs/tabs.js

```diff
@@ -774,9 +810,26 @@ class Tabs extends LocalizeCoreElement(ArrowKeysMixin(SkeletonMixin(LitElement))
 
 		if (!this._initialized && this._tabs.length === 0) return;
 
-		let selectedTab = this._tabs.find((tab) => tab.selected && tab.state !== 'removing');
+		let selectedTab = null;
+		const newTabIds = {};
+		this._tabs?.forEach((tab) => {
+			if (this._initialized && !reduceMotion && this._tabs.length !== Object.keys(this._tabIds).length) {
+				// if it's a new tab, update state to animate addition
+				if (!this._tabIds[tab.id]) {
+					this._tabIds[tab.id] = true;
+					tab.setAttribute('data-state', 'adding');
+				}
+			}
+			if (!selectedTab && tab.selected && tab.getAttribute('data-state') !== 'removing') {
+				selectedTab = tab;
+			}
+			newTabIds[tab.id] = true;
+		});
+
+		this._tabIds = newTabIds;
+
 		if (!selectedTab) {
-			selectedTab = this._tabs.find((tab) => tab.state !== 'removing');
+			selectedTab = this._tabs.find((tab) => tab.getAttribute('data-state') !== 'removing');
```

Comment:
In theory the removal would trigger a slot change and there wouldn't be any tabs with state of removing but I've kept this check and the one above just in case